src/CoreBundle/private/library/classes/dbobjects/TCMSDownloadFileEndPoint.class.php

```diff
@@ -160,46 +163,58 @@ public function &GetFileType()
      */
     public function GetDownloadLink($dummyLink = false, $bHideName = false, $bHideSize = false, $bCreateLinkWithToken = false, $bHideIcon = false, $sDownloadLinkName = '', $oExtranetUser = null)
     {
-        if (!array_key_exists('sFullURL', $this->_cacheParameter)) {
-            $sLink = $this->GetPlainDownloadLink($dummyLink, $bCreateLinkWithToken, false, $oExtranetUser);
+        return $this->getDownloadHtmlTag($dummyLink, $bHideName, $bHideSize, $bHideIcon, $sDownloadLinkName);
+    }
 
-            $mediaIdTag = '';
-            if ($dummyLink) {
-                $mediaIdTag = ' cmsdocument="'.$this->id.'"';
-            }
+    /**
+     * Renders HTML tag of download with icon and size.
+     */
+    public function getDownloadHtmlTag(
+        bool $isWysiwygBackendLink = false,
+        bool $hideName = false,
+        bool $hideSize = false,
+        bool $hideIcon = false,
+        string $downloadLinkName = ''): string
+    {
+        if (array_key_exists('downloadHtmlTag', $this->_cacheParameter)) {
+            return $this->_cacheParameter['downloadHtmlTag'];
+        }
 
-            $humanReadableFileSize = self::GetHumanReadableFileSize($this->sqlData['filesize']);
+        $viewRenderer = new \ViewRenderer();
 
-            $sName = TGlobalBase::OutHTML($this->GetName());
-            if (!empty($sDownloadLinkName)) {
-                $sName = TGlobal::OutHTML($sDownloadLinkName, false);
-            }
+        $downloadUrl = $this->GetPlainDownloadLink($isWysiwygBackendLink);
 
-            $extraStyle = '';
-            if ($bHideName) {
-                $sName = '&nbsp;';
-                $extraStyle .= 'style="text-decoration:none"';
-            }
+        $fileName = TGlobalBase::OutHTML($this->GetName());
+        if ('' !== $downloadLinkName) {
+            $fileName = TGlobal::OutHTML($downloadLinkName, false);
+        }
 
-            $sSizeHTML = '';
-            if (!$bHideSize) {
-                $sSizeHTML = '<span class="downloadsize">['.TGlobalBase::OutHTML($humanReadableFileSize).']</span>';
-            }
+        $downloadLinkDataModel = new DownloadLink($this->id, $downloadUrl, $fileName);
```

Comment:
Our current code guidelines are a bit blurry about the model class name (could be `DownloadLinkDataModel`).

But it would be sufficient to call the variable here `$downloadLink`.

Reply:
The namespace/directory already tells me it is a DataModel and in the directory is arready a dataModel without the suffix. I imho would add the suffix. But there should be a specific guideline for this.

Reply:
I added the DataModel suffix